### PR TITLE
cleanup scheduler cache informerFactory

### PR DESCRIPTION
**Pull request by xiaoanyunfei**
`informerFactory` can manage all the informers registering in it.

### CODE REVIEW

**Review by shinytang6 — COMMENTED** [review]
/hold

File: pkg/scheduler/cache/cache.go [hunk 1]
```diff
@@ -538,20 +541,7 @@ func newSchedulerCache(config *rest.Config, schedulerName string, defaultQueue s
...
-	go sc.scInformer.Informer().Run(stopCh)
-	go sc.queueInformerV1beta1.Informer().Run(stopCh)
-	go sc.quotaInformer.Informer().Run(stopCh)
-	go sc.cpuInformer.Informer().Run(stopCh)
-
-	if options.ServerOpts.EnablePriorityClass {
-		go sc.pcInformer.Informer().Run(stopCh)
-	}
-
+	sc.informerFactory.Start(stopCh)
```
Comment (shinytang6) [inline pkg/scheduler/cache/cache.go:544]: have you tested it? l think `informerFactory` doesn't contain volcano related informers?

**Review by shinytang6 — CHANGES_REQUESTED** [review]

File: pkg/scheduler/cache/cache.go (hunk 1 above)
Comment (xiaoanyunfei) [reply pkg/scheduler/cache/cache.go:546]: En, I forgot to add volcano informerFactory.
I have added it.

File: pkg/scheduler/cache/cache.go [hunk 2]
```diff
@@ -562,26 +555,9 @@ func (sc *SchedulerCache) Run(stopCh <-chan struct{}) {
...
-				sc.cpuInformer.Informer().HasSynced,
-			}
-			if options.ServerOpts.EnablePriorityClass {
-				informerSynced = append(informerSynced, sc.pcInformer.Informer().HasSynced)
-			}
-			return informerSynced
-		}()...,
-	)
+func (sc *SchedulerCache) WaitForCacheSync(stopCh <-chan struct{}) {
+	sc.informerFactory.WaitForCacheSync(stopCh)
```
Comment (shinytang6) [inline pkg/scheduler/cache/cache.go:559]: should we cope with the cacheSync error ?

Comment (xiaoanyunfei) [reply pkg/scheduler/cache/cache.go:559]: > should we cope with the cacheSync error ?

But so far no functions care about the result of `WaitForCacheSync`

I can add some code to check the result of `informerFactory.WaitForCacheSync` and return bool value,  if that is you want

**Review by shinytang6 — COMMENTED** [review]
Thanks, lgtm overall :)

File: pkg/scheduler/cache/cache.go (hunk 2 above)
Comment (shinytang6) [reply pkg/scheduler/cache/cache.go:559]: never mind, it's ok to just leave it here

**Review by shinytang6 — APPROVED** [review]
/lgtm
/approve

**Review by jasonliu747 — COMMENTED** [review]
/lgtm

**Comment by Thor-wl** [comment]
@xiaoanyunfei Thanks for your contributions. It's my honor to invite you to be member of Volcano originzation with an issue.

**Comment by jasonliu747** [comment]
> @xiaoanyunfei Thanks for your contributions. It's my honor to invite you to be member of Volcano originzation with an issue.

image

**Comment by huone1** [comment]
https://kubernetes.io/docs/concepts/storage/storage-capacity/
storage-capacity in Kubernetes v1.19 is alpha， alpha is disable by default.
So the action to  sync all  resource object  is wrong in k8s 1.19

**Pull request merged**